akka-http-cors/src/main/scala/ch/megard/akka/http/cors/scaladsl/settings/CorsSettings.scala

```diff
@@ -119,6 +122,51 @@ abstract class CorsSettings extends javadsl.settings.CorsSettings {
 
 object CorsSettings {
 
+  private val prefix = "akka.http.cors"
+
+  def apply(config: Config) = fromSubConfig(config.getConfig(prefix))
+
+  private def parseStringList(config: Config, path: String): List[String] = {
```

Comment:
This pattern is usually expressed using `recover`, for example: (change is as you feel it)
```scala
  private def parseStringList(config: Config, path: String): List[String] = {
    Try(config.getStringList(path).asScala.toList)
      .recover {
        case _: ConfigException.WrongType => config.getString(path).split(' ').toList
      }
      .get
  }

  private def parseSeconds(config: Config, path: String): Option[Long] = {
    Try(Some(config.getLong(path)))
      .recover {
        case _: ConfigException.Missing => None
        case _: ConfigException.WrongType => Some(config.getDuration(path, TimeUnit.SECONDS))
      }
      .get
  }
```

I am still not a big fan of using exceptions instead of if/else, but as the behaviour is correct, this can be addressed (or not) another time.